spec/realtime/channel.test.js

```diff
@@ -336,6 +336,359 @@ define(['ably', 'shared_helper', 'async'], function(Ably, helper, async) {
 		}
 	};
 
+	testOnAllTransports(_exports, 'attachWithChannelParamsBasicChannelsGet', function(realtimeOpts) { return function(test) {
+		test.expect(3);
+		var testName = 'attachWithChannelParamsBasicChannelsGet';
+		try {
+			var realtime = helper.AblyRealtime(realtimeOpts);
+			realtime.connection.on('connected', function() {
+				var params = {
+					modes: 'subscribe',
+					delta: 'vcdiff'
+				};
+				var channelOptions = {
+					params: params
+				};
+				var channel = realtime.channels.get(testName, channelOptions);
+				channel.attach(function(err) {
+					if(err)
+						test.ok(false, 'Attach failed with error: ' + displayError(err));
+					test.deepEqual(channel.channelOptions, channelOptions, 'Check requested channel options');
+					test.deepEqual(channel.params, params, 'Check result params');
+					test.deepEqual(channel.modes, ['subscribe'], 'Check result modes');
+					closeAndFinish(test, realtime);
+				});
+			});
+			monitorConnection(test, realtime);
+		} catch(e) {
+			test.ok(false, testName + ' failed with exception: ' + e.stack);
+			closeAndFinish(test, realtime);
+		}
+	}});
+
+	testOnAllTransports(_exports, 'attachWithChannelParamsBasicSetOptions', function(realtimeOpts) { return function(test) {
+		test.expect(3);
+		var testName = 'attachWithChannelParamsBasicSetOptions';
+		try {
+			var realtime = helper.AblyRealtime(realtimeOpts);
+			realtime.connection.on('connected', function() {
+				var params = {
+					modes: 'subscribe',
+					delta: 'vcdiff'
+				};
+				var channelOptions = {
+					params: params
+				};
+				var channel = realtime.channels.get(testName);
+				channel.setOptions(channelOptions);
+				channel.attach(function(err) {
+					if(err)
+						test.ok(false, 'Attach failed with error: ' + displayError(err));
+					test.deepEqual(channel.channelOptions, channelOptions, 'Check requested channel options');
+					test.deepEqual(channel.params, params, 'Check result params');
+					test.deepEqual(channel.modes, ['subscribe'], 'Check result modes');
```

Comment:
As above.  Also, should we consider testing what happens if you subscribe after call `setOptions`?

Reply:
Test added in https://github.com/ably/ably-js/pull/613/commits/b727b65b35ef4d05a66bc1902d886d1ab1286ce6